Adds blight:remove_timer() function.

xvxx opened the pull request:
Addresses #153.

LiquidityC reviewed (changes requested):
The timer also needs to be removed from the timer thread. Look in `timer.rs`. Currently there's an event to clear all timers. Extending this with a method to drop a specific id should be simple enough.

Also, to be safe you might change `lua_script.rs:run_timed_function()` to handle if the requested callback is nil which could happen if you're unlucky with concurrency.

xvxx commented:
@LiquidityC I'm always unlucky with concurrency, so I added that nil check and am now removing the timer from the scheduler properly (in theory).

I updated the docs with a really simple test case, but am happy to try testing some more complicated timers too.

LiquidityC reviewed (approved):
Looking good now. Thanks
